### PR TITLE
Add initial Nav2 configuration

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This PR adds the initial Nav2 bringup and other configurations for the C1T truck.

## Related Jira Key

Closes [CF-818](https://usdot-carma.atlassian.net/browse/CF-818)

## Motivation and Context

Incremental build up of Nav2-based C1T autonomy stack

## How Has This Been Tested?

Drove around in C-Hub

## Types of changes

- [x] New feature (non-breaking change that adds functionality)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.

[CF-818]: https://usdot-carma.atlassian.net/browse/CF-818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# c1t_bringup
+
+This package contains the launch and configuration files needed to bring up a C1T vehicle.
+
+## Prerequisites
+
+You must complete the following configuration steps before your C1T vehicle will be ready for bringup.
```
Comment (willjohnsonk) [inline README.md:7]: I don't know how verbose we want this to be but should packages like nav2-bringup, the vesc driver, and the sllidar driver be included in the install directions?

File: launch/navigation_launch.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<launch>
+  <node pkg="nav2_controller" exec="controller_server">
+     <param from="$(find-pkg-share c1t_bringup)/params/params.yaml"/>
+  </node>
+
+  <node pkg="nav2_smoother" exec="smoother_server">
+    <!-- <param from="$(find-pkg-share c1t_bringup)/params/params.yaml"/> -->
```
Comment (willjohnsonk) [inline launch/navigation_launch.xml]: what's the reasoning behind commenting some of these param references out?

File: launch/navigation_launch.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  <node pkg="nav2_bt_navigator" exec="bt_navigator">
+    <param from="$(find-pkg-share c1t_bringup)/params/params.yaml"/>
+  </node>
+
+  <node pkg="nav2_waypoint_follower" exec="waypoint_follower">
+    <!-- <param from="$(find-pkg-share c1t_bringup)/params/params.yaml"/> -->
+  </node>
+
+  <!-- <node pkg="nav2_velocity_smoother" exec="velocity_smoother">
```
Comment (willjohnsonk) [inline launch/navigation_launch.xml]: Any reason we weren't using velocity smoothing? Can't remember if the vesc natively handles it

File: params/params.yaml [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+  local_costmap:
+    ros__parameters:
+      update_frequency: 5.0
+      publish_frequency: 2.0
+      global_frame: odom
+      robot_base_frame: base_link
+      rolling_widow: true
+      width: 3
+      height: 3
+      resolution: 0.05
```
Comment (willjohnsonk) [inline params/params.yaml]: should be updated to 0.04 to match the map resolution

File: params/params.yaml [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+      robot_radius: 0.22
```
Comment (willjohnsonk) [inline params/params.yaml]: truck radius should be measured and added

File: params/params.yaml [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+      plugins: ["static_layer"]
+      # plugins: ["voxel_layer", "inflation_layer"]
```
Comment (willjohnsonk) [inline params/params.yaml]: should probably test with inflation_layer enabled

File: params/params.yaml [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+      always_send_full_costmap: True
+
+global_costmap:
+  global_costmap:
+    ros__parameters:
+      update_frequency: 1.0
+      publish_frequency: 1.0
+      global_frame: map
+      robot_base_frame: base_link
+      robot_radius: 0.22
```
Comment (willjohnsonk) [inline params/params.yaml]: same as above

File: params/params.yaml [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+      resolution: 0.05
```
Comment (willjohnsonk) [inline params/params.yaml]: same as above

**Review by willjohnsonk — COMMENTED** [review]
added early comments for the draft since it's still a work in progress

File: launch/navigation_launch.xml (hunk 2 above)
Comment (adamlm) [reply launch/navigation_launch.xml]: It was when I was playing around with Nav2's launch. This PR is still volatile, so these will likely be changed.

File: launch/navigation_launch.xml (hunk 3 above)
Comment (adamlm) [reply launch/navigation_launch.xml]: I'm trying to use as few nodes as possible right now so as to not over-complicate things.

File: params/params.yaml (hunk 5 above)
Comment (adamlm) [reply params/params.yaml]: Pushed to a new story: [CF-823](https://usdot-carma.atlassian.net/browse/CF-823)

File: params/params.yaml (hunk 7 above)
Comment (adamlm) [reply params/params.yaml]: Pushed to a new story: [CF-823](https://usdot-carma.atlassian.net/browse/CF-823)

File: launch/navigation_launch.xml (hunk 2 above)
Comment (adamlm) [reply launch/navigation_launch.xml]: Removed the commented-out portions.

File: launch/navigation_launch.xml (hunk 3 above)
Comment (adamlm) [reply launch/navigation_launch.xml]: Removed the commented-out portions.

File: params/params.yaml (hunk 6 above)
Comment (adamlm) [reply params/params.yaml]: Updated.

File: README.md (hunk 1 above)
Comment (adamlm) [reply README.md:7]: I think we can skip this for now. There might be a better place for those instructions.

File: package.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>c1t_bringup</name>
+  <version>0.0.0</version>
+  <description>TODO: Package description</description>
```
Comment (willjohnsonk) [inline package.xml]: Placeholders left in

File: params/params.yaml [hunk 10]
```diff
@@ -0,0 +1,172 @@
+amcl:
+  ros__parameters:
+    laser_max_range: -1.0 # Use value reported in laser scan
```
Comment (willjohnsonk) [inline params/params.yaml:3]: I'm assuming this has already been tested but this looks bizarre, any further details?

**Review by willjohnsonk — CHANGES_REQUESTED** [review]

File: params/params.yaml (hunk 10 above)
Comment (adamlm) [reply params/params.yaml:3]: Yes, see the [AMCL configuration docs](https://navigation.ros.org/configuration/packages/configuring-amcl.html).

**Review by willjohnsonk — APPROVED** [review]

**Pull request merged**